[Fix] Fix for Paldean Tauros and Lycanroc Forms have Access to Other Forms' TM Movepools

PrabbyDD opened the pull request:
## What are the changes the user will see?
Pokemon such as paldean tauros, rotom, lycanroc, etc will now have proper tms listed in shop related to current form (i.e. fan rotom will no longer be able to get blizzard as tm)

## Why am I making these changes?
Bug fix for  #4361 and #4360 

## What are the changes from a developer perspective?
fixing a bracket that was misplaced

### Screenshots/Videos

https://github.com/user-attachments/assets/c44a2500-1de8-44cd-a34e-e0ea886d2733

https://github.com/user-attachments/assets/a63a4056-4736-403c-b363-7c1ec6a7ee1a

## How to test the changes?
Override files

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- ~~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~~
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

flx-sta commented on src/field/pokemon.ts:
```diff
@@ -3972,7 +3972,7 @@ export class PlayerPokemon extends Pokemon {
       for (const p of tmSpecies[tm]) {
         if (Array.isArray(p)) {
-          if (p[0] === this.species.speciesId || (this.fusionSpecies && p[0] === this.fusionSpecies.speciesId) && p.slice(1).indexOf(this.species.forms[this.formIndex]) > -1) {
+          if ((p[0] === this.species.speciesId || this.fusionSpecies && p[0] === this.fusionSpecies.speciesId) && p.slice(1).indexOf(this.species.forms[this.formIndex]) > -1) {
```
may want to improve the readability at the same time tbh:


```suggestion
          const [pkm, ...forms] = p;
    
          if ((pkm === this.species.speciesId || this.fusionSpecies && pkm === this.fusionSpecies.speciesId) && forms.some(form => form === this.species.forms[this.formIndex])) {
```
(Please check if this properly written. I don't have code-check in the browser)

flx-sta reviewed (changes requested):
Please make sure to add some test-coverage for this change.
If you need help with writing test feel free to reach out here and/or on discord

PrabbyDD commented:
I am not really sure how to test this instance? Should I just check if rotom fan can learn blizzard?

flx-sta commented:
> I am not really sure how to test this instance? Should I just check if rotom fan can learn blizzard?

for example, yes.
Basically check that whatever wasn't working before, is working now. So if someone was to ever disable it, it get's caught by the tests

PrabbyDD commented:
I see. What folder should I put something like this in? Should I put it in the misc file?

flx-sta commented:
> I see. What folder should I put something like this in? Should I put it in the misc file?

Technically you can run `npm run create-test` but I doubt any of those types matches your use-case.
Just put it somewhere you think it should be and it can be move anytime again anyway

PrabbyDD commented:
Is it ok now? I added a specific test, which also helped me find another bug with my previous commit that I fixed (the respective forms can now learn their signature moves, like fan rotom can learn air slash, etc)

flx-sta commented on src/test/field/pokemon.test.ts:
```diff
@@ -63,4 +64,32 @@ describe("Spec - Pokemon", () => {
       });
     });
   });
+  it("pokemon that have form changes and different tms per form should not share tms between forms", async () => {
+    game.override.starterForms({ [Species.ROTOM]: 4 });
+    await game.classicMode.startBattle([Species.ROTOM]);
+    const playerPokemon = game.scene.getPlayerPokemon()!;
+
+    // 59 is blizzard, fan rotom should not be in that array
+    const compatible1 = tmSpecies[59].some(p => {
+      if (Array.isArray(p)) {
+        const [pkm, form] = p;
+        return pkm === playerPokemon.species.speciesId && playerPokemon.getFormKey() === form;
+      }
+      return false;
+    });
+
+    // Air slash is 403, fan rotom should be in it
+    const compatible2 = tmSpecies[403].some(p => {
+      if (Array.isArray(p)) {
+        const [pkm, form] = p;
+        return pkm === playerPokemon.species.speciesId && playerPokemon.getFormKey() === form;
+      }
+      return false;
+    });
+
+    expect(playerPokemon.compatibleTms.includes(59)).toBeFalsy();
+    expect(playerPokemon.compatibleTms.includes(403)).toBeTruthy();
+    expect(compatible1).toBeFalsy();
+    expect(compatible2).toBeTruthy();
+  });
```
Don't forget to import `Moves` too!

```suggestion

  it("should not share tms between different forms", async () => {
    game.override.starterForms({ [Species.ROTOM]: 4 });

    await game.classicMode.startBattle([Species.ROTOM]);

    const fanRotom = game.scene.getPlayerPokemon()!;

    expect(fanRotom.compatibleTms).not.toContain(Moves.BLIZZARD);
    expect(fanRotom.compatibleTms).toContain(Moves.AIR_SLASH);
  });
```
